x-pack/plugins/reporting/server/config/schema.ts

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema, TypeOf } from '@kbn/config-schema';
+import moment from 'moment';
+
+const KibanaServerSchema = schema.object({
+  hostname: schema.maybe(
+    schema.string({
+      validate(value) {
+        if (value === '0') {
+          return 'must not be "0" for the headless browser to correctly resolve the host';
+        }
+      },
+      hostname: true,
+    })
+  ),
+  port: schema.maybe(schema.number()),
+  protocol: schema.maybe(
+    schema.string({
+      validate(value) {
+        if (!/^https?$/.test(value)) {
+          return 'must be "http" or "https"';
+        }
+      },
+    })
+  ),
+}); // default values are all dynamic in createConfig$
+
+const QueueSchema = schema.object({
+  indexInterval: schema.string({ defaultValue: 'week' }),
+  pollEnabled: schema.boolean({ defaultValue: true }),
+  pollInterval: schema.number({ defaultValue: 3000 }),
+  pollIntervalErrorMultiplier: schema.number({ defaultValue: 10 }),
+  timeout: schema.number({ defaultValue: moment.duration(2, 'm').asMilliseconds() }),
+});
+
+const RulesSchema = schema.object({
+  allow: schema.boolean(),
+  host: schema.maybe(schema.string()),
+  protocol: schema.maybe(schema.string()),
+});
+
+const CaptureSchema = schema.object({
+  timeouts: schema.object({
+    openUrl: schema.number({ defaultValue: 30000 }),
+    waitForElements: schema.number({ defaultValue: 30000 }),
+    renderComplete: schema.number({ defaultValue: 30000 }),
+  }),
+  networkPolicy: schema.object({
+    enabled: schema.boolean({ defaultValue: true }),
+    rules: schema.arrayOf(RulesSchema, {
+      defaultValue: [
+        { host: undefined, allow: true, protocol: 'http:' },
+        { host: undefined, allow: true, protocol: 'https:' },
+        { host: undefined, allow: true, protocol: 'ws:' },
+        { host: undefined, allow: true, protocol: 'wss:' },
+        { host: undefined, allow: true, protocol: 'data:' },
+        { host: undefined, allow: false, protocol: undefined }, // Default action is to deny!
+      ],
+    }),
+  }),
+  zoom: schema.number({ defaultValue: 2 }),
+  viewport: schema.object({
+    width: schema.number({ defaultValue: 1950 }),
+    height: schema.number({ defaultValue: 1200 }),
+  }),
+  loadDelay: schema.number({
+    defaultValue: moment.duration(3, 's').asMilliseconds(),
+  }), // TODO: use schema.duration
```

Comment:
maybe good time to switch over to the `schema.duration`?

Reply:
We can do that for 8.0 only, but not backport it. It would be a breaking change: if a user had this config existing in kibana.yml, it should not break when they do a minor version upgrade.

Reply:
Gotcha, makes sense